File: Jack.Presentation/Controllers/API/FamiliaCriancaController.vb
Imports System.Web.Http
Imports Jack.Model.DTOs

Namespace Controllers.API
    Public Class FamiliaCriancaController
        Inherits ApiController
        Public Function ObterCriancas(ID As Integer) As IList(Of DTOCrianca)

            Dim lstRetorno As List(Of DTOCrianca) = Nothing
            Dim oBusiness As Business.FamiliaCrianca

            Try
                oBusiness = New Business.FamiliaCrianca()
                lstRetorno = oBusiness.ObterCriancasByFamilia(ID)

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno

        End Function


        Public Function ObterCriancasByRepresentante(ID As Integer) As IList(Of DTOCriancaRepresentante)

            Dim lstRetorno As List(Of DTOCriancaRepresentante) = Nothing
            Dim oBusiness As Business.FamiliaCrianca

            Try
                oBusiness = New Business.FamiliaCrianca()
                lstRetorno = oBusiness.ObterCriancasByFamiliaWithRep(ID)

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno

        End Function

    End Class
End Namespace

File: Jack.Presentation.Web/Controllers/API/FamiliaCrianca/FamiliaCriancaController.cs

using System;
using System.Collections.Generic;
using System.Web.Http;
using Business = Jack.Business;
using Model = Jack.Model;

namespace Controllers.API
{
	public class FamiliaCriancaController : ApiController
	{
		public IList<Model.Criancas> ObterCriancas(int ID)
		{

			IList<Model.Criancas> lstRetorno = null;
			Business.FamiliaCrianca oBusiness = default(Business.FamiliaCrianca);

            try
            {
                oBusiness = new Business.FamiliaCrianca();
                lstRetorno = oBusiness.ObterCriancasByFamilia(ID);

            }
            catch (Exception ex)
            {
                lstRetorno = null;
            }
            finally
            {
                oBusiness = null;
            }

			return lstRetorno;

		}


		public IList<Model.Criancas> ObterCriancasByRepresentante(int ID)
		{

			IList<Model.Criancas> lstRetorno = null;
			Business.FamiliaCrianca oBusiness = default(Business.FamiliaCrianca);

            try
            {
                oBusiness = new Business.FamiliaCrianca();
                lstRetorno = oBusiness.ObterCriancasByFamiliaWithRep(ID);

            }
            catch (Exception ex)
            {
                lstRetorno = null;
            }
            finally
            {
                oBusiness = null;
            }

			return lstRetorno;

		}

	}
}
